README.md
# Talos-

register/history
Rcontract = web3.eth.contract([)
{
"inputs": [
{
"internalType": "string",
"name": "key",
"type": "string"
}
],
"name": "getData",
"outputs": [
{
"internalType": "uint256",
"name": "",
"type": "uint256"
}
],
"stateMutability": "nonpayable",
"type": "function"
},
{
"inputs": [
{
"internalType": "string",
"name": "key",
"type": "string"
}
],
"name": "getOwner",
"outputs": [
{
"internalType": "string",
"name": "",
"type": "string"
}
],
"stateMutability": "nonpayable",
"type": "function"
},
{
"inputs": [
{
"internalType": "string",
"name": "key",
"type": "string"
}
],
"name": "getSensor",
"outputs": [
{
"internalType": "string",
"name": "",
"type": "string"
}
],
"stateMutability": "nonpayable",
"type": "function"
},
{
"inputs": [
{
"internalType": "string",
"name": "key",
"type": "string"
}
],
"name": "getValue",
"outputs": [
{
"internalType": "uint256",
"name": "",
"type": "uint256"
}
],
"stateMutability": "nonpayable",
"type": "function"
},
{
"inputs": [
{
"internalType": "string",
"name": "key",
"type": "string"
},
{
"internalType": "string",
"name": "owner",
"type": "string"
},
{
"internalType": "uint256",
"name": "value",
"type": "uint256"
},
{
"internalType": "string",
"name": "sensor",
"type": "string"
}
],
"name": "set",
"outputs": [],
"stateMutability": "nonpayable",
"type": "function"
}
]
Rcontract = web3.eth.contract([)
Rcontract = web3.eth.contract([ )
Rcontract = web3.eth.contract([ \)
Rcontract = web3.eth.contract([)
{
"inputs": [
{
"internalType": "string",
"name": "key",
"type": "string"
}
],
"name": "getData",
"outputs": [
{
"internalType": "uint256",
"name": "",
"type": "uint256"
}
],
"stateMutability": "nonpayable",
"type": "function"
},
{
"inputs": [
{
"internalType": "string",
"name": "key",
"type": "string"
}
],
"name": "getOwner",
"outputs": [
{
"internalType": "string",
"name": "",
"type": "string"
}
],
"stateMutability": "nonpayable",
"type": "function"
},
{
"inputs": [
{
"internalType": "string",
"name": "key",
"type": "string"
}
],
"name": "getSensor",
"outputs": [
{
"internalType": "string",
"name": "",
"type": "string"
}
],
"stateMutability": "nonpayable",
"type": "function"
},
{
"inputs": [
{
"internalType": "string",
"name": "key",
"type": "string"
}
],
"name": "getValue",
"outputs": [
{
"internalType": "uint256",
"name": "",
"type": "uint256"
}
],
"stateMutability": "nonpayable",
"type": "function"
},
{
"inputs": [
{
"internalType": "string",
"name": "key",
"type": "string"
},
{
"internalType": "string",
"name": "owner",
"type": "string"
},
{
"internalType": "uint256",
"name": "value",
"type": "uint256"
},
{
"internalType": "string",
"name": "sensor",
"type": "string"
}
],
"name": "set",
"outputs": [],
"stateMutability": "nonpayable",
"type": "function"
}
Rcontract
Rcontract = web3.eth.contract([{"constant":true,"inputs":[{"name":"key","type":"string"}],"name":"getSensor","outputs":[{"name":"","type":"string"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[{"name":"key","type":"string"}],"name":"getOwner","outputs":[{"name":"","type":"string"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":false,"inputs":[{"name":"key","type":"string"},{"name":"owner","type":"string"},{"name":"value","type":"uint256"},{"name":"sensor","type":"string"}],"name":"set","outputs":[],"payable":false,"stateMutability":"nonpayable","type":"function"},{"constant":true,"inputs":[{"name":"key","type":"string"}],"name":"getValue","outputs":[{"name":"","type":"uint256"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[{"name":"key","type":"string"}],"name":"getData","outputs":[{"name":"","type":"uint256"}],"payable":false,"stateMutability":"view","type":"function"}])
Rcontract
Registro = Rcontract.at("0x68bc7ca0eE382Ae165466c9f65606291F1a8181f")
eth.accouts[0]
eth.accounts[0]
Registro.set.sendTransaction("primero","juan",5,"Infrarrojo",{from: eth.accounts[0],})
Registro.get.call("primero")
Registro.getData.call("primero")
Registro.getOwner.call("primero")
Registro.getValue.call("primero")
Registro.getSensor.call("primero")
Registro.set.sendTransaction("primero","juan",5,"Infrarrojo",{from: eth.accounts[0],})
Registro.set.sendTransaction("primero","juan",4,"Infrarrojo",{from: eth.accounts[0],})
Registro.getValue.call("primero")
Registro.set.sendTransaction("segundo","robot2",45,"Infrarrojo",{from: eth.accounts[0]})
Registro.getValue.call("segundo")
Rcontract = web3.eth.contract([{"constant":true,"inputs":[{"name":"key","type":"string"}],"name":"getSensor","outputs":[{"name":"","type":"string"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[{"name":"key","type":"string"}],"name":"getOwner","outputs":[{"name":"","type":"string"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":false,"inputs":[{"name":"key","type":"string"},{"name":"owner","type":"string"},{"name":"value","type":"uint256"},{"name":"sensor","type":"string"}],"name":"set","outputs":[],"payable":false,"stateMutability":"nonpayable","type":"function"},{"constant":true,"inputs":[{"name":"key","type":"string"}],"name":"getValue","outputs":[{"name":"","type":"uint256"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[{"name":"key","type":"string"}],"name":"getData","outputs":[{"name":"","type":"uint256"}],"payable":false,"stateMutability":"view","type":"function"}])
Registro = Rcontract.at("0x68bc7ca0eE382Ae165466c9f65606291F1a8181f")
Registro.getValue.call("segundo")
Registro.getOwner.call("segundo")

pytalos-client/IoT.py
import serial
import time
import io
from client import *

Datoin='';



Bandera = False
cl = client("http://192.168.5.200:8545")
def Envia_BCH(Dato):
        global cl
        result = cl.call_sm(Dato)
        print("resultado: ",result)

def Recibe_valor():
        while True:
                try:
                        Datoin=str(ESP32Lora.read(3))
                        Datoin=Datoin.lstrip('b')
                        Datoin=Datoin.strip('\'')
                        if (len(Datoin))>2:
                                print(Datoin)
                                Envia_BCH(int(Datoin))
                        Datoin=''
                        ESP32Lora.reset_input_buffer()
                        ESP32Lora.reset_output_buffer()
                        break

                except KeyboardInterrupt:
                        print("Exiting")
                        break
                time.sleep(0.1)


try:
        ESP32Lora = serial.Serial('/dev/ttyUSB0',115200, timeout=1)
        time.sleep(1.8)

        ESP32Lora.reset_input_buffer()
        ESP32Lora.reset_output_buffer()


except serial.SerialException:
    print('Port is not available')
    Bandera = True

except serial.portNotOpenError:
    print('Attempting to use a port that is not open')
    print('End of script')
    Bandera = True


while True:
        if Bandera:
                print('ESP32 Lora sin conexion')
        else:
                Recibe_valor()
                time.sleep(0.2)


pytalos-client/stup.py
import setuptools

with open("README.md", "r") as fh:
    long_description = fh.read()

setuptools.setup(
    name="pytalos-client",
    version="1.0.0",
    author="Juan de Anda Suárez",
    author_email="juan.ds@purisima.tecnm.mx",
    description="pySEED is a python inplementation of symmetric-approximation energy-based estimation of distribution (seed): a continuous optimization algorithm, that was published in IEEE Accsses, with DOI: 10.1109/ACCESS.2019.2948199",
    long_description="pySEED-EDA is a Python implementation of the Symmetric-Approximation Energy-Based Estimation of Distribution (SEED) algorithm: A Continuous Optimization Algorithm , which allows the optimization in continuous space for independent variable functions, based on distribution estimation algorithms, in the Univariate Marginal Distribution scheme [2], the main idea is to make a generational change in each population evolution under the Boltzmann distribution probability function (PDF-B), because PDF-B is a function that has the property that states with less energy are unlikely, so SEED converges in each evolution to a better or equal energy state.",
    long_description_content_type="text/markdown",
    url="https://github.com/LIDT-Lab/pySEED.git",
    packages=setuptools.find_packages(),
    classifiers=[
        "Programming Language :: Python :: 3",
        "License :: OSI Approved :: MIT License",
        "Operating System :: OS Independent",
    ],
    python_requires='>=3.5.2',
    install_requires=[
   'numpy>=1.18.2'
   ]
)


client-js/app.js
var Web3 = require("web3");
var express = require("express");
var app = express();
var server = require("http").createServer(app);

web3 = new Web3(new Web3.providers.HttpProvider("http://localhost:8545"));
server.listen(8080);

app.use(express.static("public"));

app.get("/", function(req, res){
        res.sendFile(__dirname + "/public/html/index.html");
})

var hashd = "0x68bc7ca0eE382Ae165466c9f65606291F1a8181f"; //smart contract registration hash address on the private Ethereum network.
var indexSmartContract = web3.eth.contract([{"constant":true,"inputs":[{"name":"key","type":"string"}],"name":"getSensor","outputs":[{"name":"","type":"string"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[{"name":"key","type":"string"}],"name":"getOwner","outputs":[{"name":"","type":"string"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":false,"inputs":[{"name":"key","type":"string"},{"name":"owner","type":"string"},{"name":"value","type":"uint256"},{"name":"sensor","type":"string"}],"name":"set","outputs":[],"payable":false,"stateMutability":"nonpayable","type":"function"},{"constant":true,"inputs":[{"name":"key","type":"string"}],"name":"getValue","outputs":[{"name":"","type":"uint256"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[{"name":"key","type":"string"}],"name":"getData","outputs":[{"name":"","type":"uint256"}],"payable":false,"stateMutability":"view","type":"function"}]) //Index to the smart contract for registration of autonomous elements

var SmartContract = indexSmartContract.at(hashd);

SmartContract.set.sendTransaction("hola","robot2",45,"Infrarrojo",{from: web3.eth.accounts[0]})
